backends/aerospike.go

```diff
@@ -0,0 +1,61 @@
+package backends
+
+import (
+	"context"
+	as "github.com/aerospike/aerospike-client-go"
+	"github.com/golang/snappy"
+)
+
+type AerospikeConfig struct {
+	host      string
+	port      int
+	namespace string
+}
+
+type Aerospike struct {
+	config *AerospikeConfig
+	client *as.Client
+}
+
+func NewAerospikeBackend(config *AerospikeConfig) (*Aerospike, error) {
+	a := &Aerospike{}
+	a.config = config
+	client, err := as.NewClient(config.host, config.port)
+	if err != nil {
+		return nil, err
+	}
+	a.client = client
+	return a, nil
+}
+
+func (a *Aerospike) Get(ctx context.Context, key string) (string, error) {
+	asKey, err := as.NewKey(a.config.namespace, "uuid", key)
+	if err != nil {
+		return "", err
+	}
+	rec, err := a.client.Get(nil, asKey)
+	if err != nil {
+		return "", err
+	}
+	val, err := snappy.Decode(nil, rec.Bins["value"].([]byte))
```

Comment:
Do you know if the performance impact of `string(snappy.Decode([]byte(someString)))` is significant?

If it's not, then we should pull this compression into a type of middleware. For example:

``` 
func SnappyCompress(delegate Backend) Backend {
  // return some struct which implements the Backend interface by running
  // the compression algorithm on Put and Get before sending them to delegate for storage
}
``` 

Basic idea being: The compression algorithm and the storage backend are orthogonal concerns. This design couples them unnecessarily. With the structure above, the app config could let people mix & match them.

That's a pretty major win for the project... but there is some performance impact, given the string/[]byte conversion.